src/sortStudents.ts

```diff
@@ -1,16 +1,55 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyStudents = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      copyStudents.sort((a, b) => a[sortBy].localeCompare(b[sortBy]));
+      break;
+
+    case SortType.Age:
+      copyStudents.sort((a, b) => b[sortBy] - a[sortBy]);
+      break;
+
+    case SortType.Married:
+      copyStudents.sort((a, b) => Number(b[sortBy]) - Number(a[sortBy]));
```

Comment:
```suggestion
    case SortType.Age:
    case SortType.Married:
      copyStudents.sort((a, b) => Number(b[sortBy]) - Number(a[sortBy]));
```
You can combine these cases too, as they are almost the same